src/sentry/integrations/github/client.py

```diff
@@ -112,6 +113,10 @@ def create_token(self):
             },
         )
 
+    @transaction_start("GitHubClientMixin.check_file")
+    def check_file(self, repo, path):
+        return self.head_cached(path="/repos/{}/contents/{}".format(repo, path)).status_code
```

Comment:
@ceorourke I don't quite understand why we'd want to return the status code. IMO it would make more sense to return nothing here or just `True` if it succeeds.

Reply:
@MeredithAnya do you have a preference for what to return?

Reply:
I thought we wanted the status code so we can check specifically for the 404/400

Reply:
This idea, although I guess the code will change https://github.com/getsentry/sentry/pull/21449/files#diff-b517da9f8145142071b7180613fc7d96383b7213f6be500fd29baa261e81cdb4R105

Reply:
@MeredithAnya @ceorourke yea but you're checking the status code of the error, not of the actual response.

Reply:
@scefali fair point
@ceorourke can we just return:

```
return self.head_cached(path="/repos/{}/contents/{}".format(repo, path))
```

to be consistent with how the other client methods work?